src/vs/workbench/services/themes/browser/workbenchThemeService.ts

```diff
@@ -486,6 +486,7 @@ export class WorkbenchThemeService extends Disposable implements IWorkbenchTheme
 	}
 
 	private applyTheme(newTheme: ColorThemeData, settingsTarget: ThemeSettingTarget, silent = false): Promise<IWorkbenchColorTheme | null> {
+		setTheme(newTheme)
```

Comment:
theme colors not only change on `applyTheme`, but also when the color customizations change.
Best is to subscribe to `onColorThemeChange` to get all notifications